### DISCUSSION TITLE
Testing pinout and basic programs on not-working-TSynth (PCB1.2 T4.1)

### DISCUSSION THREAD

**Discussion by emilien-g** [ElectroTechnique/electrotechnique.github.io#33] [Hardware Issues]
Hi all ! 
I've built the TSynth with PCB v1.2 and Teensy 4.1, i've soldered everything except potentiometers and LEDs
I've double-checked my solder points with a lens

The result is : 
- TSynth firmware is not working (nothing seems to boot)
- Simple Teensy 4.1 example patches (like blink) work properly (including with the TSynth PCB USB plug)
- Simple Teensy 4.1 audio patches work properly (including with the TSynth PCB 6.35 stereo plug)
- Simple patches for the screen don't work
- Simple patches for USB MIDI don't work (i'm super familiar with them, I even have created several PCBs using that feature)

Anybody faced the same problem or could know where my problems come from ?
What are the recommendations to test each soldering / component / function independently to ensure that the PCB i bought has no default and the screen I received is not broken ?
In addition : with PCB 1.2, I should use firmware V2_03, right ? 

Thanks a lot !

**Comment by Electrotechnique** [comment]
Hello, you can use **all** T4.1 firmware with PCB V1.2 - the current release is V2.11. How have you obtained the firmware (did you download the entire zip file repository?) Can you monitor the serial output as you boot it?

**Reply by emilien-g** [reply]
@ElectroTechnique thanks for the answer !
Ok perfect for the firmware, I did both : I downloaded the github repo and uploaded each version with teensy loader, then compiled the code from the arduino editor ("Usb Type = Serial + MIDI + Audio", CPU = 600MHz, Optimize = fast)

The screen is still not working with even no backlight

- And with the simple USB MIDI examples from the teensy examples, i don't get anything from my MIDI keyboard plugged into the USB Host of the TSynth PCB (i did some PCBs exposing a plug for the teensy 3.6 USB Host and I never had problems)
- in the same manner, with the simple MIDI examples from the teensy examples, i don't get anything when plugging my keyboard to the DIN plug

Today i also checked my soldering points on the microscope + tested the connection with a multimeter, and everything seems to be ok on the connection side 🤨

**Reply by emilien-g** [reply]
@ElectroTechnique 
edit : I had problem with my SD card, i re-formatted it properly

serial monitor says 

"SD card is connected

Ignoring Dir

[...] all patches are printed from 1 to 64 [...]

MIDI Ch:0 (0 is Omni On)

USB HOST MIDI Class Compliant Listening

USB Client MIDI Listening

MIDI In DIN Listening

Set Patch: Solina"

**Reply by Wizards-of-Zen** [reply]
So any news on this issue? I'm At a stand still with my kit also. Same issue, the display doesn't show anything. I might see if I have anothe one to use.

**Comment by Electrotechnique** [comment]
What do you mean _uploaded each version with teensy loader, then compiled the code from the arduino editor ("Usb Type = Serial + MIDI + Audio", CPU = 600MHz, Optimize = fast)_?

You just flash the latest V2.11 firmware and nothing else. You shouldn't be compiling any code unless you're doing development work.

**Reply by emilien-g** [reply]
Thanks for the answer ! 
Yes i did it because it doesn't work, that's why...
I'm a developper and super familiar with arduino and teensy, so when seeing it didn't work and didn't even boot, I checked everything as I said (electrically and in the code), tried to find some ways to perform unit tests (teensy 4, audio board, USB Host, Screen only, Midi plugs), and yes I tested your versions by uploading and testing each firmware version, then compiling / uploading the code from the arduino IDE, and none of them work

**Reply by emilien-g** [reply]
Sorry that i didn't make it clear that my situation is that it doesn't work so i'm searching for ways to test all parts of the PCB / code in order to understand where it's broken

**Reply by Wizards-of-Zen** [reply]
Same going on with mine

**Comment by emilien-g** [comment]
Thanks for you answers @ElectroTechnique and @Wizards-of-Zen !
In the end, it was the screen that made the microcontroller crash. I ordered another one on the web, and everything worked. My original kit was from munichen-audio.de, i will write them
But i still think that it's good to have a troubleshooting procedure when someone starts a tsynth, with minimal arduino code to test each part separately (controls + serial, screen, audio out, USB programming connection...).
Do you know if anybody plan to do it ?

**Reply by Electrotechnique** [reply]
Hello, glad it now works. Resellers were advised to test their displays before they posted them, because these are low-cost and usually received without much packaging.

The code is being worked on at the moment by a couple of much better developers than myself, and has been extensively refactored. Both the envelopes and glide now have exponential characteristics, and memory usage has dropped considerably. We're hoping for another release soon, with a feature that will be very significant.

**Comment by Wizards-of-Zen** [comment]
Oh cool!
